Hazel/src/Hazel/Renderer/Renderer2D.cpp

```diff
@@ -163,12 +165,8 @@ namespace Hazel {
 
 	void Renderer2D::FlushAndReset()
 	{
-		EndScene();
-
-		s_Data.QuadIndexCount = 0;
-		s_Data.QuadVertexBufferPtr = s_Data.QuadVertexBufferBase;
-
-		s_Data.TextureSlotIndex = 1;
+		Flush();
+		StartBatch();
 	}
```

Comment:
Since you added a function `StartBatch()`, wouldn't it make sense to rename this to something along the lines of `FlushAndRestartBatch()` or simply `NextBatch()`?

Reply:
I like the sound of NextBatch()! I renamed the method and updated the PR.